modules/merkleIdSystem.js

```diff
@@ -8,42 +8,80 @@
 import * as utils from '../src/utils.js'
 import {ajax} from '../src/ajax.js';
 import {submodule} from '../src/hook.js'
-import { getStorageManager } from '../src/storageManager.js';
+import {getStorageManager} from '../src/storageManager.js';
 
 const MODULE_NAME = 'merkleId';
-const SESSION_COOKIE_NAME = '_svsid';
 const ID_URL = 'https://id2.sv.rkdms.com/identity/';
+const DEFAULT_REFRESH = 7 * 3600;
+const SESSION_COOKIE_NAME = '_svsid';
 
 export const storage = getStorageManager();
 
 function getSession(configParams) {
   let session = null;
-  if (typeof configParams.sv_session !== 'string') {
+  if (typeof configParams.sv_session === 'string') {
     session = configParams.sv_session;
   } else {
-    session = readCookie() || readFromLocalStorage();
+    session = storage.getCookie(SESSION_COOKIE_NAME);
   }
   return session;
 }
 
-function readCookie() {
-  return storage.cookiesAreEnabled() ? storage.getCookie(SESSION_COOKIE_NAME) : null;
+function setCookie(name, value, expires) {
+  let expTime = new Date();
+  expTime.setTime(expTime.getTime() + expires * 1000 * 60);
+  storage.setCookie(name, value, expTime.toGMTString());
 }
 
-function readFromLocalStorage() {
-  return storage.localStorageIsEnabled() ? storage.getDataFromLocalStorage(SESSION_COOKIE_NAME) : null;
+function setSession(storage, response) {
+  utils.logInfo('Merkle setting session ');
+  if (response && response.c && response.c.value && typeof response.c.value === 'string') {
+    setCookie(SESSION_COOKIE_NAME, response.c.value, storage.expires);
+  }
 }
 
-function constructUrl(configParams) {
-  const session = getSession(configParams);
+function constructUrl(configParams, storedId) {
+  const session = getSession(configParams, storedId);
```

Comment:
This 2nd parameter hasn't been added to the `getSession` function

Reply:
Fixed

Reply:
@msm0504 mind approving your suggested changes have been implemented?